cmd/register/register_util.go

```diff
@@ -437,4 +474,88 @@ func getFlyteTestManifest() ([]FlyteSnack, string, error) {
 		return nil, "", err
 	}
 	return FlyteSnacksRelease, *releases[0].TagName, nil
+
+}
+
+func getRemoteStoragePath(ctx context.Context, s *storage.DataStore, remoteLocation, file, identifier string) (storage.DataReference, error) {
+	remotePath, err := s.ConstructReference(ctx, storage.DataReference(remoteLocation), fmt.Sprintf("%v-%v", identifier, file))
+	if err != nil {
+		return storage.DataReference(""), err
+	}
+	return remotePath, nil
+}
+
+func uploadFastRegisterArtifact(ctx context.Context, file, sourceCodeName, version string) error {
+	dataStore, err := getStorageClient(ctx)
+	if err != nil {
+		return err
+	}
+	var dataRefReaderCloser io.ReadCloser
+	remotePath := storage.DataReference(rconfig.DefaultFilesConfig.SourceUploadPath)
+	if len(rconfig.DefaultFilesConfig.SourceUploadPath) == 0 {
+		remotePath, err = dataStore.ConstructReference(ctx, dataStore.GetBaseContainerFQN(ctx), "fast")
+		if err != nil {
+			return err
+		}
+	}
+	rconfig.DefaultFilesConfig.SourceUploadPath = string(remotePath)
+	fullRemotePath, err := getRemoteStoragePath(ctx, dataStore, rconfig.DefaultFilesConfig.SourceUploadPath, sourceCodeName, version)
+	if err != nil {
+		return err
+	}
+	raw, err := json.Marshal(file)
+	if err != nil {
+		return err
+	}
+	dataRefReaderCloser, err = os.Open(file)
+	if err != nil {
+		return err
+	}
+	dataRefReaderCloser, err = gzip.NewReader(dataRefReaderCloser)
+	if err != nil {
+		return err
+	}
+	if err := dataStore.ComposedProtobufStore.WriteRaw(ctx, fullRemotePath, int64(len(raw)), storage.Options{}, dataRefReaderCloser); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getStorageClient(ctx context.Context) (*storage.DataStore, error) {
+	if Client != nil {
+		return Client, nil
+	}
+	testScope := promutils.NewTestScope()
+	// Set Keys
+	labeled.SetMetricKeys(contextutils.AppNameKey, contextutils.ProjectKey, contextutils.DomainKey)
+	s, err := storage.NewDataStore(storage.GetConfig(), testScope.NewSubScope("flytectl"))
+	if err != nil {
+		logger.Errorf(ctx, "error while creating storage client %v", err)
+		return Client, err
+	}
+	Client = s
+	return Client, nil
+}
+
+func isFastRegister(file string) bool {
+	f := strings.Split(file, "/")
+	if strings.HasPrefix(f[len(f)-1], "fast") && strings.HasSuffix(f[len(f)-1], sourceCodeExtension) {
```

Comment:
i think we should add this as a comment to the package, how we determine what is fast